### PR TITLE
start port over to the shared kzg lib

### PR DESCRIPTION
**Pull request by roberto-bayardo**
There is still code to be moved from Prysm into the shared lib, but this should at least update Prysm fully to the latest EIP-4844 spec changes.  All devnet e2e tests pass when running with the similarly [upgraded geth](https://github.com/mdehoog/go-ethereum/pull/49).

### CODE REVIEW

**Comment by roberto-bayardo** [comment]
cc: @terencechain

File: beacon-chain/core/blob/sidecar.go [hunk 1]
```diff
@@ -1,143 +1,57 @@
...
-	b, err := kzg.VerifyKZGProof(bytesutil.ToBytes48(aggregatedPolyCommitment), &x, &y, bytesutil.ToBytes48(sidecar.AggregatedProof))
-	if err != nil {
-		return err
-	}
-	if !b {
-		return errors.New("couldn't verify proof")
-	}
-	return nil
+	copy(kzgSidecar.BeaconBlockRoot[:], sidecar.BeaconBlockRoot)
+	copy(kzgSidecar.KZGAggregatedProof[:], sidecar.AggregatedProof)
```
Comment (Inphi) [inline beacon-chain/core/blob/sidecar.go]: nit: could simplify these using `bytesutil.ToBytes48(proof)` on L48

**Review by Inphi — APPROVED** [review]
the deps.bzl file needs to be updated too using gazelle. I can do this later.

File: beacon-chain/core/blob/sidecar.go (hunk 1 above)
Comment (roberto-bayardo) [reply beacon-chain/core/blob/sidecar.go]: nice, changed.

**Comment by roberto-bayardo** [comment]
> the deps.bzl file needs to be updated too using gazelle. I can do this later.

looking into this.

**Comment by roberto-bayardo** [comment]
> > the deps.bzl file needs to be updated too using gazelle. I can do this later.
> 
> looking into this.

Ran the updates, and patched various bazel build issues. "bazel build //beacon-chain:beacon-chain" now runs successfully.

**Pull request merged**